### PR TITLE
Final implementation of v1.0

### PR DESCRIPTION
**Pull request by mi-char**
BREAKING CHANGE: Removed Stor client implementation
BREAKING CHANGE: Upgrade to the latest version of Cats, Htt4s, Pureconfig, ...
BREAKING CHANGE: HCP backend requires authenticated access
BREAKING CHANGE: Removed build-in Future support
feat: HCP backend is balancing load by random IP address selection
feat: Removed Monix dependency

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ language: java
 scala:
 - 2.12.4
 jdk:
-- oraclejdk8
+- oraclejdk9
```
Comment (sideeffffect) [inline .travis.yml]:
```suggestion
- openjdk11
```
? or however we get to OpenJDK 11

File: build.gradle [hunk 2]
```diff
@@ -7,14 +7,15 @@ buildscript {
...
 
 ext {
-    circeVersion = "0.9.3"
-    metricsVersion = "2.4.4"
-    http4sVersion = "0.18.15"
-    monixVersion = "3.0.0-RC1"
+    circeVersion = "0.13.0"
+    metricsVersion = "2.6.6"
+    http4sVersion = "0.21.1"
+    monixVersion = "3.1.0"
```
Comment (jendakol) [inline build.gradle]: Please add note this is used only in tests.

File: README.md [hunk 3]
```diff
@@ -66,21 +65,3 @@ val backend: StorageBackend[Task] = ???
...
-import monix.eval.Task
-import monix.execution.Scheduler
-
-implicit val scheduler: Scheduler = ???
-implicit val taskToFuture: Task ~> Future = ???
-
-val backend: StorageBackend[Task] = ???
-
-val client: StorageClient[Future] = StorageClient[Task](backend).mapK[Future]
-```
```
Comment (jendakol) [inline README.md]: Did you really remove the support for this? If so, why there's still `fkTaskToFuture` present in tests?

File: build.gradle (hunk 2 above)
Comment (mi-char) [reply build.gradle]: Done. Added comment.

File: README.md (hunk 3 above)
Comment (mi-char) [reply README.md]: Originally I've just removed the documentation part, but after your comment, I've completely removed the Future support from the library.

**Pull request merged**